### PR TITLE
[JBPM-9983] Allow to define number of Retries for WIH exception handl…

### PR DESCRIPTION
**Pull request by abhijithumbe**
…ing strategy

**JIRA**: _(https://issues.redhat.com/browse/JBPM-9983)_ 

**referenced Pull Requests**: _(https://github.com/kiegroup/droolsjbpm-knowledge/pull/593)_

### CODE REVIEW

File: jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
     protected String handlingProcessId;
     protected String handlingStrategy;
 
-    protected int retries = 3;
+    protected int retries = Integer.parseInt(System.getProperty("org.jbpm.exception.handlingStrategy.retry.count", "3"));
```
Comment (afalhambra) [inline jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java:40]: Just a small suggestion. Most (if not all) of our system properties are not using camel case, so I would use something like 
```org.jbpm.exception.handling_strategy.retry.count```
But it's just a suggestion

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/handler/WorkItemHandlerExceptionHandlingTest.java [hunk 2]
```diff
@@ -221,6 +222,7 @@ public void testErrornousHandlerWithStrategyMultipleRetry() throws Exception {
         ProcessInstance processInstance = ksession.startProcess("com.sample.boolean", params);
         assertEquals(ProcessInstance.STATE_COMPLETED, processInstance.getState());
         assertProcessVarValue(processInstance, "isChecked", "true");
-        assertEquals(3, workItemHandler.getCounter());
+        assertEquals(4, workItemHandler.getCounter());
```
Comment (afalhambra) [inline jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/handler/WorkItemHandlerExceptionHandlingTest.java:225]: Please, clean up the system property `org.jbpm.exception.handlingStrategy.retry.count` already defined above. You could either use a `finally` block or remove it with the `After` junit method.

**Review by afalhambra — CHANGES_REQUESTED** [review]
Overall looks good, just some minor changes

File: jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java (hunk 1 above)
Comment (abhijithumbe) [reply jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java]: Updated system property as per suggestion.

**Review by afalhambra — APPROVED** [review]
Thank you @abhijithumbe - great work!

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by abhijithumbe** [comment]
Thanks @afalhambra  :)

**Comment by afalhambra** [comment]
Jenkins retest this

**Pull request merged**